smart_open/smart_open_lib.py

```diff
@@ -269,6 +276,34 @@ def __exit__(self, type, value, traceback):
         self.read_key.close()
 
 
+class SSHOpenRead(object):
+    """
+    Implement streamed reader from remote server using SSH
+
+    """
+    def __init__(self, parsed_uri):
+        if parsed_uri.scheme not in ("ssh"):
+            raise TypeError("can only process remote files via ssh")
+        self.parsed_uri = parsed_uri
+
+    def __iter__(self):
+        ssh = subprocess.Popen("ssh -o StrictHostKeyChecking=no " + self.parsed_uri.netloc + " 'cat " + self.parsed_uri.uri_path + "'",
```

Comment:
Probably better (more robust) to pass a list of arguments to `Popen` ctr, rather than a single hand-assembled string.